### PR TITLE
Allow users to specify negative preferences

### PR DESCRIPTION
**Pull request by naomiajacobs**
- Adds a preference type column to the `MoviePreference` model with a migration (will need to run `flask db upgrade` after this gets merged
- Adds a route for updating `MoviePreferences`
- Displays the type with emojis (✅ or 🔴 ) and lets you toggle by clicking

Drawbacks: still very slow to get user feedback once you toggle, will work on that in a follow-up PR.

### CODE REVIEW

File: app/routes.py [hunk 1]
```diff
@@ -67,6 +67,12 @@ def create_movie_preference(user):
     return MoviePreferenceController().create(user)
 
 
+@app.route('/users/<user_id>/movie-preferences/<movie_preference_id>', methods=['PUT'])
```
Comment (norseboar) [inline app/routes.py]: I think this should be a PATCH, PUT is a full replacement update I think

https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/PATCH

File: src/components/MovieList.tsx [hunk 2]
```diff
@@ -17,7 +18,15 @@ export default function MovieList(props: {
       My movies:
       {props.movies.map((movie, index) => (
         <div className="movieOption" key={index}>
-          {movie.title}
+          <div className="movieOptionWrapper">
```
Comment (norseboar) [inline src/components/MovieList.tsx]: OOC why can't this all be in the main movieOption div? It seems like a bit more to keep track of

File: src/components/MovieList.tsx [hunk 3]
```diff
@@ -17,7 +18,15 @@ export default function MovieList(props: {
...
+            <MoviePreferenceType
+              movie_preference_id={movie.id}
```
Comment (norseboar) [inline src/components/MovieList.tsx]: I think we want to camelCase all js names?

File: src/components/MoviePreferenceType.tsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+import { updateMoviePreference } from "../network/requests";
+import { PreferenceType, SetMovies, User } from "../types";
+
+export default function MoviePreferenceType({
+  movie_preference_id,
+  preference,
+  setMovies,
+  user
+}: {
+  movie_preference_id: number;
```
Comment (norseboar) [inline src/components/MoviePreferenceType.tsx]: See above re: camelCase

File: src/network/requests.ts [hunk 5]
```diff
@@ -46,3 +46,23 @@ export const deleteMovie = (
     .then(response => response.json())
     .then(() => fetchMovies(user, setMovies));
 };
+
+export const updateMoviePreference = async (
+  id: number,
+  user: User,
+  preference_type: PreferenceType,
```
Comment (norseboar) [inline src/network/requests.ts]: camelCase again

File: src/network/requests.ts [hunk 6]
```diff
@@ -46,3 +46,23 @@ export const deleteMovie = (
...
+) => {
+  const response = await fetch(`users/${user.id}/movie-preferences/${id}`, {
+    method: "PUT",
+    headers: {
+      "Content-Type": "application/json; charset=utf-8"
+    },
+    body: JSON.stringify({ preference_type: preference_type })
+  });
+  if (response.status !== 200) {
+    alert(`Failed to update movie preference.'`);
```
Comment (norseboar) [inline src/network/requests.ts]: This can just be a string, doesn't need to have the backticks

File: src/components/MovieList.tsx (hunk 2 above)
Comment (naomiajacobs) [reply src/components/MovieList.tsx]: styling with flexbox to get the emoji and title on the left, and the x on the right

File: app/routes.py (hunk 1 above)
Comment (naomiajacobs) [reply app/routes.py]: huh TIL! I had always seen them used interchangeably

File: app/controllers/movie_preference_controller.py [hunk 7]
```diff
@@ -27,6 +43,11 @@ def get(self, user):
             movie_preference_dict.update({"title": external_movie["title"]})
             movies.append(movie_preference_dict)
 
+        def sort_by_title(movie):
+            return movie["title"]
+
+        movies.sort(key=sort_by_title)
```
Comment (zenovy) [inline app/controllers/movie_preference_controller.py]: Could we do this in a lambda?

**Pull request merged**